ext/src/helpers/output_limited_buffer.rs

```diff
@@ -21,8 +75,12 @@ impl OutputLimitedBuffer {
     }
 }
 
-impl io::Write for OutputLimitedBuffer {
-    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+impl OutputLimitedBuffer {
+    fn check_write(&mut self) -> StreamResult<usize> {
+        Ok(usize::MAX)
```

Comment:
The caller will invoke `check_write` and then invoke `write` with a buffer that fits in the `usize` returned. If it repeatedly returns 0, then the caller will hang indefinitely so this shouldn't return 0.